File: solutions/number_of_1_bits.java
/**
 * Approach/Algorithm:
 * The solution counts the number of 1 bits in the binary representation of an integer.
 * It uses a loop to check each bit of the number until the number becomes zero.
 * Time Complexity: O(1) - The loop runs a maximum of 32 times (for 32-bit integers)
 * Space Complexity: O(1) - Uses a constant amount of extra space
 */
class Solution {
    public int hammingWeight(int n) {
        ArrayList<Integer> arr = new ArrayList<>();

        int var = n;

        while (var != 0) {

        }

File: solutions/number_of_1_bits.js
/**
 * Approach/Algorithm: The solution counts the number of 1 bits in the binary representation of a number.
 * It does this by repeatedly checking the least significant bit (using n & 1) and right-shifting the number.
 * Time Complexity: O(1) - The loop runs a maximum of 32 times (for 32-bit integers)
 * Space Complexity: O(1) - Uses constant extra space
 */
var hammingWeight = function(n) {
    let count = 0;
    while (n !== 0) {
        count += n & 1; // Check if the least significant bit is 1
        n = n >>> 1;    // Right shift to check the next bit
    }
    return count;
};